Google.Api.Generator/CodeGenerator.cs

```diff
@@ -75,24 +75,34 @@ internal static class CodeGenerator
             { typeof(Xunit.Assert).Namespace, "xunit" },
         };
 
-        private static readonly IReadOnlyList<string> AllowedAdditionalServices = new List<string>
-        {
-            IAMPolicy.Descriptor.FullName,
-            Locations.Descriptor.FullName,
-            Operations.Descriptor.FullName,
-        }.AsReadOnly();
-
         /// <summary>
         /// For unaliased source file context, we still have to sometimes aliase some namespaces to avoid
         /// name collisions.
         /// This collection is used to influence the order in which colliding namespaces are picked for
         /// aliasing.
         /// </summary>
-        private static readonly IReadOnlyCollection<Regex> s_avoidAliasingNamespaceRegex = new HashSet<Regex>
+        private static readonly IReadOnlyCollection<Regex> AvoidAliasingNamespaceRegex = new HashSet<Regex>
         {
             new Regex(@"^System\.?.*", RegexOptions.Compiled | RegexOptions.CultureInvariant),
         };
 
+        /// <summary>
+        /// For unaliased source file context, forces the alias on namespaces on listed on value
+        /// when generating the package in key.
+        /// This is useful for when a package depends on another (not mixin) and we know there are name collisions.
+        /// </summary>
+        private static readonly IReadOnlyDictionary<string, IReadOnlyCollection<string>> ForceAliasForLibrary = new Dictionary<string, IReadOnlyCollection<string>>
+        {
+            { "Google.Cloud.Kms.Inventory.V1", new List<string> { "Google.Cloud.Kms.V1" } }
```

Comment:
Presumably we *could* make this part of configuration eventually.

Reply:
Yes, +100 to that.